src/matter/session/SessionManager.ts

```diff
@@ -59,11 +60,38 @@ export class SessionManager<ContextT> {
         return this.sessions.get(sessionId);
     }
 
+    destroySession(sessionId: number) {
+        const session = this.sessions.get(sessionId);
+        if (session) {
+            session.destroy();
+            this.sessions.delete(sessionId);
+        }
+    }
+
+    destroySessionForFabricIndex(fabricIndex: FabricIndex) {
+        [...this.sessions.values()].forEach(session => {
+            if (!session.isSecure()) return false;
+            const secureSession = session as SecureSession<any>;
+            if (secureSession.getFabric()?.fabricIndex.index === fabricIndex.index) {
+                session.destroy();
+                this.sessions.delete(session.getId());
+            }
+        });
+    }
+
+    destroyAllSessions() {
```

Comment:
This should never be called: UNICAST_UNSECURE_SESSION_ID pointing to unsecureSession is added in the constructor and this code will delete it.

What is the purpose of this method? I am not sure I understand when all the sessions need to be destroyed

Reply:
you are right ... I remive it again